src/coreclr/vm/assembly.cpp

```diff
@@ -598,16 +603,18 @@ Assembly *Assembly::CreateDynamic(AppDomain *pDomain, CreateDynamicAssemblyArgs
     }
 
     NewHolder<DomainAssembly> pDomainAssembly;
+    BOOL                      createdNewAssemblyLoaderAllocator = FALSE;
 
     {
         GCX_PREEMP();
 
+        AssemblyLoaderAllocator* pAssemblyLoaderAllocator = nullptr;
```

Comment:
Nit: Maybe declare it separately in each of the branches. It's not used after the if, so ideally it should not leak out.

Reply:
Fixed